File: db4oj.tests/src/com/db4o/db4ounit/common/assorted/StaticFieldUpdateConsistencyTestCase.java
/* Copyright (C) 2010  Versant Inc.   http://www.db4o.com */
package com.db4o.db4ounit.common.assorted;

import com.db4o.*;
import com.db4o.config.*;
import com.db4o.consistency.*;

import db4ounit.*;
import db4ounit.extensions.*;

public class StaticFieldUpdateConsistencyTestCase extends AbstractDb4oTestCase {

	public static class SimpleEnum {
		public String _name;
		
		public SimpleEnum(String name) {
			_name = name;
		}
		
		public static SimpleEnum A = new SimpleEnum("A");
		public static SimpleEnum B = new SimpleEnum("B");
	}
	
	public static class Item {
		public SimpleEnum _value;
		
		public Item(SimpleEnum value) {
			_value = value;
		}
	}

	
	@Override
	protected void configure(Configuration config) throws Exception {
		config.updateDepth(5);
		config.objectClass(SimpleEnum.class).persistStaticFieldValues();
	}

	@Override
	protected void store() throws Exception {
		store(new Item(SimpleEnum.A));
	}
	
	public void test() throws Exception {
		updateAll();
		ConsistencyReport consistencyReport = new ConsistencyChecker(fileSession()).checkSlotConsistency();
		Assert.isTrue(consistencyReport.consistent(), consistencyReport.toString());
	}

	private void updateAll() {
		ObjectSet<Item> result = newQuery(Item.class).execute();
		while(result.hasNext()) {
			Item item = result.next();
			item._value = (item._value == SimpleEnum.A) ? SimpleEnum.B : SimpleEnum.A;
			store(item);
		}
		commit();
	}
	
}

File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Assorted/StaticFieldUpdateConsistencyTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4oUnit;
using Db4oUnit.Extensions;
using Db4objects.Db4o;
using Db4objects.Db4o.Config;
using Db4objects.Db4o.Consistency;
using Db4objects.Db4o.Tests.Common.Assorted;

namespace Db4objects.Db4o.Tests.Common.Assorted
{
	public class StaticFieldUpdateConsistencyTestCase : AbstractDb4oTestCase
	{
		public class SimpleEnum
		{
			public string _name;

			public SimpleEnum(string name)
			{
				_name = name;
			}

			public static StaticFieldUpdateConsistencyTestCase.SimpleEnum A = new StaticFieldUpdateConsistencyTestCase.SimpleEnum
				("A");

			public static StaticFieldUpdateConsistencyTestCase.SimpleEnum B = new StaticFieldUpdateConsistencyTestCase.SimpleEnum
				("B");
		}

		public class Item
		{
			public StaticFieldUpdateConsistencyTestCase.SimpleEnum _value;

			public Item(StaticFieldUpdateConsistencyTestCase.SimpleEnum value)
			{
				_value = value;
			}
		}

		/// <exception cref="System.Exception"></exception>
		protected override void Configure(IConfiguration config)
		{
			config.UpdateDepth(5);
			config.ObjectClass(typeof(StaticFieldUpdateConsistencyTestCase.SimpleEnum)).PersistStaticFieldValues
				();
		}

		/// <exception cref="System.Exception"></exception>
		protected override void Store()
		{
			Store(new StaticFieldUpdateConsistencyTestCase.Item(StaticFieldUpdateConsistencyTestCase.SimpleEnum
				.A));
		}

		/// <exception cref="System.Exception"></exception>
		public virtual void Test()
		{
			UpdateAll();
			ConsistencyReport consistencyReport = new ConsistencyChecker(FileSession()).CheckSlotConsistency
				();
			Assert.IsTrue(consistencyReport.Consistent(), consistencyReport.ToString());
		}

		private void UpdateAll()
		{
			IObjectSet result = NewQuery(typeof(StaticFieldUpdateConsistencyTestCase.Item)).Execute
				();
			while (result.HasNext())
			{
				StaticFieldUpdateConsistencyTestCase.Item item = ((StaticFieldUpdateConsistencyTestCase.Item
					)result.Next());
				item._value = (item._value == StaticFieldUpdateConsistencyTestCase.SimpleEnum.A) ? 
					StaticFieldUpdateConsistencyTestCase.SimpleEnum.B : StaticFieldUpdateConsistencyTestCase.SimpleEnum
					.A;
				Store(item);
			}
			Commit();
		}
	}
}
